pydicom/tag.py

```diff
@@ -1,20 +1,36 @@
-"""Define Tag class to hold a DICOM (group, element) tag."""
-# Copyright (c) 2008-2012 Darcy Mason
-# This file is part of pydicom, released under a modified MIT license.
-#    See the file LICENSE included with this distribution, also
-#    available at https://github.com/pydicom/pydicom
-
-# Store the 4 bytes of a dicom tag as an arbitary length integer
-#      (python "long" in python <3; "int" for python >=3).
-# NOTE: This must be not be stored as a tuple internally, as some code logic
-#       (e.g. in write_AT of filewriter) checks if a value is a multi-value
+# Copyright 2008-2017 pydicom authors. See LICENSE file for details.
+"""Define Tag class to hold a DICOM (group, element) tag and related functions.
+
+The 4 bytes of the DICOM tag are stored as an arbitrary length 'int'. Tags are
+stored as a single number and separated to (group, element) as required.
+"""
+# NOTE: Tags must be not be stored as a tuple internally, as some code logic
+#       (e.g. in filewriter.write_AT) checks if a value is a multi-value
 #       element
-# So, represent as a single number and separate to (group, element) when
-#       necessary.
+from contextlib import contextmanager
 
 from pydicom import compat
 
 
+@contextmanager
+def tag_in_exception(tag):
+    """Use `tag` within a context.
+
+    Used to include the tag details in the traceback message when an exception
+    is raised within the context.
+
+    Parameters
+    ----------
+    tag : pydicom.tag.Tag
+        The tag to use in the context.
+    """
+    try:
+        yield
+    except Exception as ex:
+        msg = 'With tag {0} got exception: {1}'.format(tag, str(ex))
+        raise type(ex)(msg)
+
+
 def Tag(arg, arg2=None):
```

Comment:
I would also prefer to use straight try/catch blocks instead of this one - apart from being more explicit, it also makes debugging easier